bundle/edu.gemini.itc/src/main/java/edu/gemini/itc/gnirs/Gnirs.java

```diff
@@ -250,11 +293,14 @@ public double getCentralWavelength() {
     }
 
     public TransmissionElement getGratingOrderNTransmission(int order) {
-        return GnirsGratingsTransmission.getOrderNTransmission(_grating, order);
+        return GnirsGratingsTransmission.getOrderNTransmission(_grating.get(), order);
     }
 
     private double correctedCentralWavelength() {
-        if (!isXDispUsed()) {
+        if (_mode instanceof Imaging) {
+            _Filter = Filter.fromFile(getPrefix(), params.filter().get().name(), getDirectory() + "/");
+            return (int) _Filter.getEffectiveWavelength();
```

Comment:
Wouldn't the conversion to `(int)` make it loose precision? The method is returning `double`